src/pages/Defi/hooks/useFoxyBalances.tsx

```diff
@@ -1,58 +1,55 @@
-import { ethChainId, toAccountId } from '@shapeshiftoss/caip'
-import { KnownChainIds } from '@shapeshiftoss/types'
-import { useEffect, useMemo, useState } from 'react'
-import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { ethChainId, fromAccountId, toAccountId } from '@shapeshiftoss/caip'
+import { useMemo } from 'react'
 import { useWallet } from 'hooks/useWallet/useWallet'
 import { useWalletSupportsChain } from 'hooks/useWalletSupportsChain/useWalletSupportsChain'
 import type { MergedFoxyOpportunity } from 'state/apis/foxy/foxyApi'
 import { useGetFoxyAprQuery, useGetFoxyBalancesQuery } from 'state/apis/foxy/foxyApi'
+import { selectPortfolioAccountsGroupedByNumberByChainId } from 'state/slices/portfolioSlice/selectors'
+import { useAppSelector } from 'state/store'
 
 export type UseFoxyBalancesReturn = {
   opportunities: MergedFoxyOpportunity[]
   totalBalance: string
   loading: boolean
 }
 
-export function useFoxyBalances() {
-  const [userAddress, setUserAddress] = useState<string | null>(null)
-
+export function useFoxyBalances({ accountNumber }: { accountNumber: number }) {
```

Comment:
This is the best we can do with the current logic given the flow of `<AccountDropdown` and DeFi balance hooks in general:

- We don't have enumeration by accountId / accountNumber yet
- The `accountId` gets set on an initial call to `onAccountIdChange` in `<AccountDropdown`
- Without this callback fired, there's no accountId, and without accountId, there's no accountNumber / no query being fired to fetch a default opportunity that will trigger an initial rendering of `<FoxyOverview />` (or any other specific step we open the modal at)

This is effectively a catch 22 where we need to *pass* an accountNumber before we can *get* an accountNumber.

Not exactly the most ergonomic for now to pass an accountNumber 0 everywhere, but this is *actually* temporary and will come in handy as we add enumeration, i.e programmatic deep-routing / balance-fetching of FOXy balances at a specific AccountId